fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java

```diff
@@ -133,10 +133,13 @@ private void schedule() {
                     try {
                         commitCompaction(partition, job,
                                 taskResult == CompactionTask.TaskResult.PARTIAL_SUCCESS /* forceCommit */);
-                        assert job.transactionHasCommitted();
+                        if (!job.transactionHasCommitted()) { // should not happen
```

Comment:
what does this `should not happen` means? If it happens, just log error message and continue? or handle it properly no matter it should happen or not?

Reply:
test tell people what to expect

Reply:
it's already handled